src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -978,16 +977,290 @@ void CmdSketcherViewSection::activated(int iMsg)
+    bool newState = !hGrp->GetBool("GridSnap");
+
+    auto* sketchView = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
+    if (sketchView) {
+        hGrp->SetBool("GridSnap", newState);
+        sketchView->GridSnap.setValue(newState);
+    }
+    return newState;
+}
+
+CmdSketcherGridSnap::CmdSketcherGridSnap()
+    : Command("Sketcher_GridSnap")
+{
+    sAppModule = "Sketcher";
+    sGroup = "Sketcher";
+    sMenuText = QT_TR_NOOP("Grid Snap");
+    sToolTipText = QT_TR_NOOP("New points will snap to the nearest grid line.\nPoints must be set closer than a fifth of the grid size to a grid line to snap.");
+    sWhatsThis = "Sketcher_GridSnap";
+    sStatusTip = sToolTipText;
+    eType = 0;
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General");
+    sPixmap = hGrp->GetBool("GridSnap") ? "Sketcher_GridSnap" : "Sketcher_GridSnap_Deactivated";
 }
 
+void CmdSketcherGridSnap::activated(int iMsg)
+{
+    Q_UNUSED(iMsg);
+    bool newState = toggleGridSnap(getActiveGuiDocument());
+    getAction()->setIcon(Gui::BitmapFactory().iconFromTheme(newState ? "Sketcher_GridSnap" : "Sketcher_GridSnap_Deactivated"));
+}
+
+bool CmdSketcherGridSnap::isActive()
+{
+    return isEditModeToolActive(getActiveGuiDocument());
+}
+
+/* Grid Auto */
+DEF_STD_CMD_A(CmdSketcherGridAuto)
+
+bool toggleGridAuto(Gui::Document* doc) {
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General");
+    bool newState = !hGrp->GetBool("GridAuto");
+
+    auto* sketchView = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
+    if (sketchView) {
+        hGrp->SetBool("GridAuto", newState);
+        sketchView->GridAuto.setValue(newState);
+    }
+    return newState;
+}
+
+CmdSketcherGridAuto::CmdSketcherGridAuto()
+    : Command("Sketcher_GridAuto")
+{
+    sAppModule = "Sketcher";
+    sGroup = "Sketcher";
+    sMenuText = QT_TR_NOOP("Grid Auto Resize");
+    sToolTipText = QT_TR_NOOP("Resize grid automatically depending on zoom.");
+    sWhatsThis = "Sketcher_GridAuto";
+    sStatusTip = sToolTipText;
+    eType = 0;
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General");
+    sPixmap = hGrp->GetBool("GridAuto") ? "Sketcher_GridAuto" : "Sketcher_GridAuto_Deactivated";
+}
+
+void CmdSketcherGridAuto::activated(int iMsg)
+{
+    Q_UNUSED(iMsg);
+    bool newState = toggleGridAuto(getActiveGuiDocument());
+    getAction()->setIcon(Gui::BitmapFactory().iconFromTheme(newState ? "Sketcher_GridAuto" : "Sketcher_GridAuto_Deactivated"));
+}
+
+bool CmdSketcherGridAuto::isActive()
+{
+    return isEditModeToolActive(getActiveGuiDocument());
+}
+
+/* Grid comp */
+DEF_STD_CMD_ACL(CmdSketcherCompGrid)

```

Comment:
In this command you more or less copy & paste the implementation of the three separate commands. It might be better to make this class derive from GroupCommand and instantiate and add the separate commands in the constructor of CmdSketcherCompGrid. As a reference implementation have a look at StdCmdLinkSelectActions

Reply:
Yes you are right. I didn't knew of GroupCommand which is what I needed indeed. But then the grid size also needs to be an individual command right?
And can the individual commands activate/inactivate other commands? For instance:  GridAuto disable the grid size.

Reply:
Also the individual commands have icons, which I replaced by checkboxes in the comp because it looked better. So I'm not sure if this can be done with the action GroupCommand. 
From what I see in StdCmdLinkSelectActions it can't.

Reply:
> And can the individual commands activate/inactivate other commands? For instance: GridAuto disable the grid size.

Yes, this is possible. The general way is to access the Application's command manager and access the command by name. GroupCommand also has an overloaded addCommand() that adds the command to a local list but there currently isn't a nice way to access it afterwards. I will extend the API accordingly by adding `getCommand(std::size_t) const`